File: BackEnd/src/main/java/org/hwadee/backend/entity/Course.java
package org.hwadee.backend.entity;

import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.LocalDate;

/**
 * 课程实体类
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Course {
    /**
     * 课程ID
     */
    private Long courseId;

    /**
     * 课程名称
     */
    private String courseName;

    /**
     * 课程编码
     */
    private String courseCode;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 授课教师ID
     */
    private Long instructorId;

    /**
     * 课程描述
     */
    private String description;

    /**
     * 课程大纲
     */
    private String syllabus;

    /**
     * 学时
     */
    private Integer creditHours;

    /**
     * 学分值
     */
    private BigDecimal creditValue;

    /**
     * 最大学生数
     */
    private Integer maxStudents;

    /**
     * 当前学生数
     */
    private Integer currentStudents;

    /**
     * 开课日期
     */
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;

    /**
     * 结课日期
     */
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;

    /**
     * 状态：0-关闭，1-开放，2-满员
     */
    private Integer status;

    public static final int CLOSED = 0;
    public static final int OPEN = 1;
    public static final int FULL = 2;

    /**
     * 创建时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
} 

File: FrontEnd/src/types/course.ts
// 课程接口
export interface Course {
  courseId: number
  courseName: string
  courseCode: string
  categoryId: number
  instructorId: number
  description: string
  syllabus: string
  creditHours: number
  creditValue: number
  maxStudents: number
  currentStudents: number
  startDate: string
  endDate: string
  status: number
  createTime?: string
  updateTime?: string
}

// 课程章节接口
export interface CourseChapter {
  chapterId: number
  courseId: number
  chapterName: string
  chapterOrder: number
  description?: string
  sections: CourseSection[]
}

// 课程小节接口
export interface CourseSection {
  sectionId: number
  chapterId: number
  sectionName: string
  resourceType: string
  contentUrl?: string
  duration?: number
  status: string
  progress: number
}

// 课程资源接口
export interface CourseResource {
  resourceId: number
  courseId: number
  resourceName: string
  resourceType: string
  fileUrl?: string
  fileSize?: number
  duration?: number
  downloadCount: number
  createTime?: string
}

// 学习记录接口
export interface LearningRecord {
  recordId: number
  userId: number
  courseId: number
  sectionId: number
  progress: number
  duration: number
  lastPosition: number
  createTime: string
  updateTime: string
}

// 测试问题选项接口
export interface TestQuestionOption {
  value: string
  text: string
}

// 测试问题接口
export interface TestQuestion {
  questionId: number
  questionText: string
  type: string
  options?: TestQuestionOption[]
  answer?: string | string[] | boolean
}

// 测试接口
export interface CourseTest {
  testId: number
  testName: string
  testType: string
  courseId: number
  startTime: string
  endTime: string
  duration: number
  questionCount: number
  totalScore: number
  status: string
  score?: number
  questions?: TestQuestion[]
}

// 分页结果接口
export interface PageResult<T> {
  list: T[]
  total: number
  page: number
  size: number
} 